docs/adr/005-git-commit-structure.md

```diff
@@ -0,0 +1,143 @@
+- Do code archaeology to see which banner (and which developer) introduced
+    a certain feature in
+    a component or a certain change in a theme (nor for blaming them, but
+    for resolving questions). The best way to do this, is using the commit
+    history.
+- "Revive" an old banner (using it as a starting point for a new one).
+    This use case is one of the reasons we introduced the `archive`
+    folder, because it's easier to copy files from a folder (using `cp` or
+    the IDE) than to copy files from a git tag. In cases where the
+    original feature branch is not yet merged, we branch off the previous
+    feature branch.
+- Develop several banners for the same channel in parallel. This happens
+    when banner requirements come in quick succession, but there are
+    change requests and iterations for the previous feature branches.
+    Since we only want to commit code that matches our quality standards,
+    we keep iterating on the code while other team members work on new
+    banner in a new feature branch.
+- Keep backwards compatibility when adding new feature to code in `src` or
+    changing a theme.
+    The `archive` folder helps with that, because we can look in the IDE for usages.
+    In the future, we might introduce more thorough automated tests and
+    checks. Keeping strict backwards compatibility is a consequence of the
+    "Revive an old banner" use case, where we want the old
+    `banners/CHANNELNAME` to work with all the intermediate fixes and
+    improvements in `src`.
+- The `banners/CHANNELNAME` structure makes it easy to navigate to the
+    banner for the current task in the file tree of the IDE/editor/CLI.
+- ??? - Question for the team: Do you see additional "use
+    cases"/scenarios where you rely on file structure or version control
+    during the campaign?
+
+### Pain points
+
+The commits to `banner/CHANNELNAME` folder work best if they consist of a
+consecutive, sequential progression of A/B tests, each test branching off
+`main` after the last test was merged. During the 2023 campaign we had
+several instances where that was not the case and banner feature branches
+branched off other banner feature branches. This led to the following
+problems:
+
+
+In a scenario where we can't branch off `main` branch for a new banner in
+the same channel, the workflow hits some snags:
+
+- We get merge and rebase conflicts when we force-push to an un-merged
+    earlier branch. It's hard to rebase the "later" branch and resolve its
+    conflicts.
+- If we merge branches in the wrong order (say Test_24, to able to work on
+    Test_25, while Test_23 is still unmerged), it becomes impossible to
+    merge earlier tests, because they would appear in the wrong order in
+    the succession of merged branches and merging would also resolve the
+    wrong conflicts.
+
+These scenarios have happened several times during the 2023 campaign.
+
+## Decision
+
+### Option 1: Get rid of "current banner" concept
```

Comment:
We already point the config file to the banner entrypoints, so having a structure like `banners/desktop/C24_blah_01, banners/desktop/C24_blah_02` would be an easy enough change.

It would also remove the archive step (and the archive), the prepare campaign step would be duplicating and renaming the banner folder for the next test. Way less merge issues because different banner branches would be working on code in different places so not confusing git so much.

The downside would be a lot of folders in the busy banner channels towards the end of the campaign, but I can deal with that. I'd prefer it over having to update a symlink every test.